lib/auth/v4/awsURIencode.ts

```diff
@@ -44,8 +44,9 @@ export default function awsURIencode(
     if (typeof input !== 'string') {
         return encoded;
     }
-    for (let i = 0; i < input.length; i++) {
-        const ch = input.charAt(i);
+    const charArray = Array.from(input);
```

Comment:
Here is what I came up with:
```js
    /**
     * Duplicate query params are not suppported by AWS S3 APIs. These params
     * are parsed as Arrays by Node.js HTTP parser which breaks this method
    */
    if (typeof input !== 'string') {
        return '';
    }

    // precalc slash and star based on configs
    const slash = encodeSlash !== undefined && encodeSlash ? '%2F' : '/';
    const star = noEncodeStar === undefined || noEncodeStar ? '*' : '%2A';
    const encoded: string[] = [];

    const charArray = Array.from(input);
    for (let ch of charArray) {
        switch (true) {
            case ch >= 'A' && ch <= 'Z':
            case ch >= 'a' && ch <= 'z':
            case ch >= '0' && ch <= '9':
            case ch === '-':
            case ch === '_':
            case ch === '~':
            case ch === '.':
                encoded.push(ch);
                break;
            case ch === '/':
                encoded.push(slash);
                break;
            case ch === '*':
                encoded.push(star);
                break;
            case ch === ' ':
                encoded.push('%20');
                break;
            default:
                encoded.push(_toHexUTF8(ch));
                break;
        }
    }
    return encoded.join('');
```

Reply:
Thanks a lot Ronnie, I added it with this commit: https://github.com/scality/Arsenal/pull/1975/commits/e9db7da105a898a198590a9fe0b450f52085b504